### PR TITLE
fix: add micro app template

### PR DESCRIPTION
**Pull request by achaljain**
Added micro app setup

 **TODO**
_In short -- Sync up micro-app template with base template_

- Merge micro app template with other templates
- Added TS setup/babel setup from the base template, remove duplicate babel dependencies if not required
- Merge src/components, src/styles from the base ( currently, base overrides the micro app template )

### CODE REVIEW

**Comment by rakeshongithub** [comment]
I think we should use TurboRepo folder structure so multiple micro apps can be created.

File: templates/microApp/package.json [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+  "main": "index.js",
+  "scripts": {
+    "build": "webpack --mode production",
+    "build:dev": "webpack --mode development",
+    "build:start": "cd dist && PORT=8080 npx serve",
+    "start": "webpack serve --open --mode development",
+    "start:live": "webpack serve --open --mode development --liveReload"
+  },
+  "devDependencies": {
+    "@babel/core": "7.13.15",
```
Comment (vinodloha) [inline templates/microApp/package.json:14]: @achaljain for dependencies like linting etc we should be using TurboRepo config package

File: templates/microApp/src/app.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/** Root Component */
+
+import React, { useState, useEffect } from 'react';
```
Comment (vinodloha) [inline templates/microApp/src/app.jsx:3]: Can we have this basic code in TS?

File: templates/microApp/src/components/atoms/Heading/index.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
```
Comment (vinodloha) [inline templates/microApp/src/components/atoms/Heading/index.js:1]: Let this be coming from common part of generator.

File: templates/microApp/src/index.html [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Document</title>
+</head>
+<body>
+  <div id="app"></div>
+  <script>
+    window.onload = () => {
+      const LIB_NAME = 'remoteApp'
```
Comment (vinodloha) [inline templates/microApp/src/index.html:12]: Can we not do a constant replacement here, based on whats the app name given to generator?

**Review by vinodloha — APPROVED** [review]

**Pull request merged**